### PR TITLE
Add documentation for MapboxDirectionsCLI tool

### LINKED ISSUE #585: Add documentation for MapboxDirectionsCLI
With the recent enhancements for the MapboxDirectionsCLI tool, we should add documentation for its use.

### Background

Recently, several tickets (#476, #540, #570) have been underway to enhance the MapboxDirectionsCLI to make it more generally useful for developers. The tool was previously able to round-trip an arbitrary, JSON-formatted Directions or Map Matching API response through model objects and back to JSON. This functionality enabled us to more easily test integration with the API in a headless environment. With these enhancements, we can convert between Directions API requests and `DirectionsOptions`, output GPX-formatted responses that can be used to simulate routes in Xcode, and provide a Directions API request URL as a fallback when no `DirectionsOptions` are provided. To put this new functionality to use, we should add documentation for increased visibility and ease of use.

### Notes and Next Steps

To make the MapboxDirectionsCLI tool more accessible for developers, we should add relevant information for its setup and potential use cases with recipes for the corresponding functionality. This documentation can be available in a markdown file linked in the existing [MapboxDirectionsCLI section](https://github.com/mapbox/mapbox-directions-swift#directions-cli) of the repository's readme.

/cc: @mapbox/navigation-ios

### PR DESCRIPTION
**Pull request by jill-cardamon**
This PR fixes #585 by adding documentation for the MapboxDirectionsCLI tool.

### CODE REVIEW

File: MapboxDirections.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -687,6 +688,7 @@
 				8A41B0FC24F5C2390021FFDC /* CHANGELOG.md */,
 				438BFEBC233D7FA900457294 /* Package.swift */,
 				438BFEC0233D805500457294 /* README.md */,
+				C318CBE626E1B29400E42936 /* MapboxDirectionsCLI.md */,
```
Comment (1ec5) [inline MapboxDirections.xcodeproj/project.pbxproj:691]: Perhaps CommandLineTool.md would be more discoverable.

File: MapboxDirectionsCLI.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
+#  Mapbox Directions CLI
+
+## Getting Started
+`MapboxDirectionsCLI` is a command line tool, designed to round-trip an arbitrary, JSON-formatted Directions or Map Matching API response through model objects and back to JSON. This is useful for various scenarios including testing purposes and designing more sophisticated API response processing pipelines. It is supplied as a Swift package.
```
Comment (1ec5) [inline MapboxDirectionsCLI.md:4]: “MapboxDirectionsCLI” isn’t a name we seem to be using anywhere from a user perspective.

```suggestion
`mapbox-directions-swift` is a command line tool, designed to round-trip an arbitrary, JSON-formatted Directions or Map Matching API response through model objects and back to JSON. This is useful for various scenarios including testing purposes and designing more sophisticated API response processing pipelines. It is supplied as a Swift package.
```

By the way, this description will need to be revisited as part of #576, which makes the tool into a more straightforward command-line version of normal usage of the library.

File: README.md [hunk 3]
```diff
@@ -217,6 +217,8 @@ To run (and build if it wasn't yet) `MapboxDirectionsCLI` and see usage:
 
 1. `swift run mapbox-directions-swift -h`
 
+For further details, see the [MapboxDirectionsCLI documentation](ADD LINK).
```
Comment (1ec5) [inline README.md:220]: Subject to the suggestion above:

```suggestion
For further details, see the [MapboxDirectionsCLI documentation](MapboxDirectionsCLI.md).
```

File: MapboxDirectionsCLI.md (hunk 2 above)
Comment (jill-cardamon) [reply MapboxDirectionsCLI.md:4]: I'll add a note to that pr as a reminder.

**Review by 1ec5 — APPROVED** [review]

**Pull request merged**